ophyd/areadetector/plugins.py

```diff
@@ -632,6 +643,26 @@ class ROIPlugin(PluginBase):
                   doc='Minimum size of the ROI in XYZ',
                   default_read_attrs=('min_x', 'min_y', 'min_z'))
 
+    def set(self, region):
+        ''' This functions allows for the ROI regions to be set.
+
+        This function takes in an ROI_number, and a dictionary of tuples and sets the ROI region.
+
+        PARAMETERS
+        ----------
+        region: dictionary.
+            A dictionary defining the region to be set, which has the structure:
+            { 'x':[min, size], 'y':[min, size], 'z':[min, size]}. 
+            Any of the keywords can be ommitted, and they will be ignored.
+        '''
+        if region is not None:
+            status = []
+            for direction, value in region.items():
```

Comment:
Re-uppping a previous comment that may have been swallowed by GitHub's UI:

The extra space after `append(` breaks PEP8. I imagine you put it there in a reasonable effort to make this long line easier to parse. Another way to do that, and reduce repetition, would be:

```py
status0 = getattr(___).set(value[0])
status1 = getattr(___).set(value[1])
statuses.extend([status0, status1])
```

Also remember the earlier comment that `put` needs to be changed to `set`. "Put" is the low-level "Just send the EPICS command," which is ultimately at the bottom of "set" as well, but "set" adds additional machinery to track done-ness.

Reply:
^ The only really important comment here is `put` -> `set`.